### DISCUSSION TITLE
Pass value to content style property in pseudo element

### DISCUSSION THREAD

**Discussion by javierrey** [vanilla-extract-css/vanilla-extract#1002] [Q&A] [answered]
Hi I need to pass a number to a shopping cart icon with a number on a badge.
The style defined in typescript file `MyComponent.css.ts` looks like this (simplified):
```
selectors: {
    ['&::after']: {
        content: updatedValue // this value must be passed at rendering time
    }
}
```
Is there a way to pass the required value from the component when it's rendered, i.e. not when buit?

If I import a dummy value at build time it shows the value in the badge correctly, but that's not valid when the value is defined in the component at the time of being rendered.

Edit: I ended up using a different approach that doesn't require the `content` property, but I'm still interested in knowing if this is possible with vanilla-extract.

<img width="33" alt="Screenshot 2023-02-17 at 13 00 46" src="https://user-images.githubusercontent.com/11460539/219659774-508ff633-093b-41f6-9d6c-1fa5a77a58f0.png">

**Comment by askoufis** [comment, marked as answer]
You could use a css variable combined with [`assignInlineVars`](https://vanilla-extract.style/documentation/packages/dynamic/#assigninlinevars). Something like this:

```ts
// MyComponent.css.ts
import { createVar } from '@vanilla-extract/css';

export const contentVar = createVar();

export const shoppingCartIcon = style({
  selectors: {
    ['&:after']: {
      content: contentVar
    }
  }
});

// MyComponent.tsx
import * as styles from "./MyComponent.css";
import { assignInlineVars } from "@vanilla-extract/dynamic";

const MyComponent = () => {
  const content = getShoppingCartItems();

  return (
    <div
      className={styles.shoppingCartIcon}
      style={assignInlineVars({ [styles.contentVar]: content })}
    />
  );
};


```

**Comment by javierrey** [comment]
Thanks Adam, that's a great code example.
